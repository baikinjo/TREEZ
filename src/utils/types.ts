export type INVENTORY_TYPE = {
  routeName: string
  id: string
  title: string
  items: {
    id: number
    description: string
    name: string
    price: string
    quantity: string
  }[]
}[]

export type ITEM_TYPE = {
  routeName: string
  id: string
  title: string
  items: {
    id: number
    description: string
    name: string
    price: string
    quantity: string
  }[]
}

export interface CART_ITEM_TYPE {
  id: number
  name: string
  price: string | number
  quantity: string | number
}

export type USER_TYPE = {
  id?: string
  displayName?: string
  email?: string
  createdAt?: Date
  orders?: {
    productId: number
    product: string
    price: number
    quantity: number
    orderAt: Date
  }[]
}
