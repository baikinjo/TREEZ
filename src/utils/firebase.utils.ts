import firebase from 'firebase/app'
import 'firebase/firestore'
import 'firebase/auth'

const config = {
  apiKey: 'AIzaSyAlJ1uc_GA8WZeF-9FWYIOlcZ-cPlL4CS0',
  authDomain: 'treez-db-baik.firebaseapp.com',
  databaseURL: 'https://treez-db-baik.firebaseio.com',
  projectId: 'treez-db-baik',
  storageBucket: 'treez-db-baik.appspot.com',
  messagingSenderId: '1412534576',
  appId: '1:1412534576:web:853bd992c457958bd3f8f5',
  measurementId: 'G-4DFZJNYYJL'
}

export const createUserProfileDocument = async (
  userAuth: firebase.User | null,
  additionalData?: any
) => {
  if (!userAuth) return

  const userRef = firestore.doc(`users/${userAuth.uid}`)
  const snapShot = await userRef.get()

  if (!snapShot.exists) {
    const { displayName, email } = userAuth
    const createdAt = new Date()

    try {
      await userRef.set({
        displayName,
        email,
        createdAt,
        ...additionalData
      })
    } catch (error) {
      console.log('error creating user', error.message)
    }
  }

  return userRef
}

export const addCollections = async (collectionKey: string, items: any) => {
  const collectionRef = firestore.collection(collectionKey)

  const batch = firestore.batch()
  items.forEach((i: any) => {
    const newDocRef = collectionRef.doc()
    batch.set(newDocRef, i)
  })

  return await batch.commit()
}

export const convertSnapshotToMap = (
  inventory: firebase.firestore.QuerySnapshot
) => {
  const transformedInventory = inventory.docs.map(
    (doc: firebase.firestore.QueryDocumentSnapshot) => {
      const { title, items } = doc.data()

      return {
        routeName: encodeURI(title.toLowerCase()),
        id: doc.id,
        title,
        items
      }
    }
  )
  return transformedInventory
}

firebase.initializeApp(config)

export const auth = firebase.auth()
export const firestore = firebase.firestore()

const provider = new firebase.auth.GoogleAuthProvider()
provider.setCustomParameters({ prompt: 'select_account' })

export const signInWithGoogle = () => auth.signInWithPopup(provider)

export default firebase
