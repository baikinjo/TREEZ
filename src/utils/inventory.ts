// @ts-ignore
export const INVENTORY_DATA = [
  {
    id: 1,
    title: 'Inventory',
    routeName: 'inventory',
    items: [
      {
        id: 1,
        description:
          'An everyday jar of flower, for relaxing after a day at work.',
        name: '3.5g Zkittlez Buds',
        price: '35.00',
        quantity: '123.00'
      },
      {
        id: 2,
        description: '500mg glass cart',
        name: '500mg cartridge',
        price: '27.75',
        quantity: '16.00'
      },
      {
        id: 3,
        description: 'Bulk flower sold by the gram, bagged on demand.',
        name: 'Bulk flower',
        price: '5.00',
        quantity: '2103.57'
      },
      {
        id: 4,
        description: 'Crunchy, and satisfying',
        name: '10mg 1:1 Chocolate Chip cookie',
        price: '8.00',
        quantity: '27.00'
      },
      {
        id: 5,
        description: 'Top quality rolling papers, made from unbleached paper',
        name: 'FUNKYMONKEY ROLLING PAPERS (6-PACK)',
        price: '2.00',
        quantity: '73.00'
      },
      {
        id: 6,
        description: 'Sized for sharing',
        name: '28g OG Kush Buds, bagged',
        price: '238.15',
        quantity: '21.00'
      },
      {
        id: 7,
        description:
          "A clear, tasteless infusion of THC that's ready to mix with your favorite cocktail. Great for birthday parties, after-work mixers, and any holiday occassion. Come see why we're everyone's #1 favorite gift this season!",
        name: 'Simply Pure, 100mg bottle',
        price: '76.00',
        quantity: '31.00'
      },
      {
        id: 8,
        description: 'Crunchy, and satisfying',
        name: '10mg 1:1 Chocolate Chip cookie (6-pack)',
        price: '45.00',
        quantity: '43.00'
      },
      {
        id: 9,
        description:
          'Infused with kief and shatter, this preroll will always take you there.',
        name: 'Mr. Sinz 1g infused preroll',
        price: '18.50',
        quantity: '88.00'
      },
      {
        id: 10,
        description:
          'Apply to skin and let stand for 10 minutes. Non-comedogenic, never tested on animals.',
        name: '1:1 50mg Relax Your Back Balm',
        price: '92.00',
        quantity: '43.00'
      }
    ]
  }
]
