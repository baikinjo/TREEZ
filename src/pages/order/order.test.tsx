import React from 'react'
import { shallow } from 'enzyme'

import OrderPage from './order'
import { USER_TYPE } from '../../utils/types'

describe('Order Page', () => {
  let wrapper: any
  beforeEach(() => {
    const currentUser: USER_TYPE = {}

    wrapper = shallow(<OrderPage currentUser={currentUser} />)
  })

  it('should render the Order Page component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
