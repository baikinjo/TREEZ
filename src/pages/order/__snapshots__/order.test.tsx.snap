// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Page should render the Order Page component 1`] = `ShallowWrapper {}`;
